shacl12-rules/index.html

```diff
@@ -700,163 +655,292 @@ <h3>RDF Rules Syntax</h3>
+                        μ(v) = μ1(v) if v in dom(μ1)
+                        μ(v) = μ2(v) if v in dom(μ2) }</pre>
+            </div>
+            <div class="ednote">
+              <p>
+              Say the domain is `dom(S1) &cups; dom(S2)`.
+              </p>
+              <p>
+                Say that two solutions that have no variables in common are compatible.
+              </p>
+            </div>
+            
+
+          </dd>
 
-      <p>The inferred triples does not include any triple present in the set of triples of G.</p>
+        </dl>
+      </section>
 
       <section id="eval-expression">
-        <h4>Evaluation of an expression</h4>
+        <h4>Evaluation of an Expression</h4>        
+
+        <p class=ednote>Sketch</p>
 
 <pre class="algorithm">
 
 Let F(arg1, arg2, ...) be an expression.
 where arg1, arg2 are RDF terms.
+@@
 
-Let [x/row] be
-    if x is an RDF term, the [x/row] is x
-    if x is a variable then [x/row] is the value of x in the row
+Let [x/<var>μ</var>] be
+    if x is an RDF term, then [x/row] is x
+    if x is a variable, then [x/row] is <var>μ</var>(x)
     ## By well-formedness, it is an error if x is not in the row.
 
-eval(F(expr1, expr2), row) = F(eval(expr1, row), eval(expr2, row))
-eval(FF(expr1, expr2) , row) = ... things that are not functions like IF
+eval(F(expr1, expr2 ...), row) = F(eval(expr1, row), eval(expr2, row), ...)
+ ...
+eval(FF(expr1, expr2) , row) = ... things that are not functions like `IF`
 </pre>
       </section>
 
       <section id="eval-rule">
-        <h4>Evaluation of a rule</h4>
+        <h4>Evaluation of a Rule</h4>
+
+        <div class=ednote>
+          <p>Sketch</p>
+          <p>Turn into definitions of the step components</p>
+        </div>
+
+
 <pre class="algorithm">
 let R be a well-formed rule.
-let rule R = (head, body) where
+let rule R = (H, B) where
              H is the sequence of triple templates in the head
              B is the sequence of triple patterns, condition expressions,
                 and assignments in the body
 
 let R : map variable to RDF term as a set of pairs (variable, RDF term)
 
-# "solution"
-let T :list of B
+## Solution sequence of one solution that does not map any variables.
+## (This is the join identity)
+
+let SEQ: Solution sequence = { <var>μ</var><sub>0</sub> }
+let G = <a>evaluation graph</a>
+
+# Start::
 
-Initial T = empty list
+Replace each blank node in B with a fresh 
+variable that does not occur in the rule body.
 
 # Evaluate rule body
-for each rule element rElt:
-    if rElt is a triple pattern:
-        T1 = empty list
-    for each row in T:
-      T1 = empty list
-      for each matches m set of (var,term)
-          row1 = row union (var, term)
-          add row1 to T1
-          endfor
-    T = T1
-    endif
+for each rule element rElt in B
+    let S1 = { }
+
+    if rElt is a triple pattern TP:
+        X = graphMatch(G, TP)
+        SEQ1 = { <var>μ</var><sub>0</sub> }
```

Comment:
isn't SEQ1 the destination bucket for the results of the join between graph match and SEQ? 

what happens if the loops fail to find any matches? wouldnt the algo then pass the identity mapping forward instead of the empty set?

Reply:
Fix: `SEQ1 = { }` (It's SEQ that is μ0 initially)

`RULE { :s :p :o } WHERE {}` generates `:s :p :o`.
`DATA` is syntactic sugar.